File: scripts/verify-governance.py
#!/usr/bin/env python3
"""本地 governance 自检（与 CI governance.yml 同逻辑的快检）"""
import pathlib, re, sys
root = pathlib.Path(__file__).resolve().parents[1]
ok = True
# 检查 Issue 模板含 Owner/priority/class。
# 豁免：config.yml（表单配置非模板）；test_feedback.yml（30 秒轻量反馈表，
# 治理字段由分诊时补——上游 dotnet-deepseek-harness-desktop 同场景长期红着没人修）。
for p in (root / ".github/ISSUE_TEMPLATE").glob("*.yml"):
    if p.name in ("config.yml", "test_feedback.yml"):
        continue
    t = p.read_text()
    for kw in ["owner", "priority", "class"]:
        if kw not in t.lower():
            print(f"FAIL {p}: 缺 {kw}")
            ok = False
# 检查 PR 模板含 checklist
pr = root / ".github/pull_request_template.md"
if pr.exists():
    t = pr.read_text()
    if "Reviewer Checklist" not in t:
        print("FAIL PR template 缺 Reviewer Checklist")
        ok = False
    if "change-scope" not in t:
        print("FAIL PR template 缺 change-scope")
        ok = False
else:
    print("FAIL 缺 .github/pull_request_template.md")
    ok = False
print("OK" if ok else "FAIL")
sys.exit(0 if ok else 1)

File: scripts/verify-governance.ts
#!/usr/bin/env -S ../../node_modules/.bin/tsx
/**
 * Local governance self-check (same logic as the CI governance job).
 *
 * - Issue templates must contain owner/priority/class.
 *   Exempt: config.yml (form config, not a template); test_feedback.yml
 *   (30s lightweight feedback form, triage fills governance fields later).
 * - PR template must contain the Reviewer Checklist + change-scope pointer.
 */
import { existsSync, readFileSync, readdirSync } from "node:fs";
import { join } from "node:path";

const root = new URL("..", import.meta.url).pathname.replace(/\/$/, "");

function main(): number {
	let ok = true;
	const issueDir = join(root, ".github/ISSUE_TEMPLATE");
	if (existsSync(issueDir)) {
		for (const name of readdirSync(issueDir)) {
			if (!name.endsWith(".yml")) continue;
			if (name === "config.yml" || name === "test_feedback.yml") continue;
			const t = readFileSync(join(issueDir, name), "utf-8");
			for (const kw of ["owner", "priority", "class"]) {
				if (!t.toLowerCase().includes(kw)) {
					console.log(`FAIL ${name}: 缺 ${kw}`);
					ok = false;
				}
			}
		}
	}
	const pr = join(root, ".github/pull_request_template.md");
	if (existsSync(pr)) {
		const t = readFileSync(pr, "utf-8");
		if (!t.includes("Reviewer Checklist")) {
			console.log("FAIL PR template 缺 Reviewer Checklist");
			ok = false;
		}
		if (!t.includes("change-scope")) {
			console.log("FAIL PR template 缺 change-scope");
			ok = false;
		}
	} else {
		console.log("FAIL 缺 .github/pull_request_template.md");
		ok = false;
	}
	console.log(ok ? "OK" : "FAIL");
	return ok ? 0 : 1;
}

process.exitCode = main();
